### ISSUE TITLE
export: Add option to extend the default lat/longs

### ISSUE THREAD

**Issue by victorlin** [nextstrain/augur#1465]
### Context

Currently, `--lat-longs` overrides the default mappings provided by [augur/data/lat_longs.tsv](https://github.com/nextstrain/augur/blob/-/augur/data/lat_longs.tsv).

### Description

Add an option to extend the default mappings in addition to the existing behavior of replacing them.

For entries in both, the extending file should take precedence over the default.

### Why?

The current behavior has encouraged curation of lat/long values independently across pathogen repos, rather than in a centralized place. Due to increased attention during the SARS-CoV-2 pandemic, the ncov workflow has a carefully curated lat/longs file. A recent PR #1449 moved all that effort into Augur's default mapping, however at this point a significant amount of pathogen repos already have their own lat/longs files.

### Proposed solutions

These are based on my speculation that most pathogen repos would benefit from extending rather than overriding – I could be wrong.

#### 1. Switch behavior directly

1. Make `--lat-longs` extend instead of replace defaults in a major release.

#### 2. Switch behavior with deprecation

1. Add an option `--lat-longs-extend` with this new additive behavior.
2. Link the current option `--lat-longs` to a new name `--lat-longs-override` with a deprecation warning that the behavior of `--lat-longs` will switch to `--lat-longs-extend` in the future.
3. Switch `--lat-longs` behavior to `--lat-longs-extend` in a major version release.

**Comment by j23414** [comment]
`+ 1` for `most pathogen repos would benefit from extending rather than overriding`

**Comment by jameshadfield** [comment]
+1 for Appending user-provided data onto the defaults. This is how `--colors` works.

What's the downside to switching the behaviour of `--lat-longs` in a single step? For a repo with their own lat-longs file then there will be no change _unless_ they are missing some which will now be filled in by the defaults, and the only reason this wouldn't be desirable is if they are using missing lat-longs as a hack to avoid plotting certain demes on the map. I'd suggest changing the behaviour of `--lat-longs` now (with a major version release).

**Comment by victorlin** [comment]
@jameshadfield good idea, I've added that as the first solution in the issue description.

**Comment by joverlee521** [comment]
I was going to revisit this issue because of the need to add custom lat/longs for rabies ([Slack thread](https://bedfordlab.slack.com/archives/CFEU0GNNS/p1736384654817249?thread_ts=1736375470.477489&cid=CFEU0GNNS)).

Unless I'm misunderstanding the issue, I think the `--lat-longs` option already works as expected.
Within `read_lat_longs`, the defaults are added to the coordinates dict, then the overrides are added:
https://github.com/nextstrain/augur/blob/19308f0a5a393c8f9b6585eb34f4aeb0a99dac93/augur/utils.py#L475-L484

For a concrete example, mpox has a custom [lat_longs.tsv](https://github.com/nextstrain/mpox/blob/02f660a2616bede159ede64a0ee8edf86a27702f/phylogenetic/defaults/lat_longs.tsv) that does not have an entry for Thailand. However, the build has [sequences from Thailand](https://nextstrain.org/mpox/clade-IIb@2025-01-01?c=clade_membership&f_country=Thailand&p=grid) that have the correct lat/longs to display them on the map.

**Comment by victorlin** [comment]
Oh! I don't remember what caused me to write up this issue, but it's possible that I only looked at the help text which says ["overrides built in mappings"](https://github.com/nextstrain/augur/blob/19308f0a5a393c8f9b6585eb34f4aeb0a99dac93/augur/export_v2.py#L967) and didn't actually test the behavior. Maybe something like this help text would be more accurate?

> Latitudes and longitudes for geography traits. See the file augur/data/lat_longs.tsv for the format. This is also the default set of latitudes and longitudes. An additional file specified by this option will extend the default set. Duplicates based on the first two columns will be resolved by taking the coordinates from the user-provided file.

**Comment by victorlin** [comment]
We could also go through [existing lat/longs](https://github.com/search?q=org%3Anextstrain+path%3A**%2Flat_longs.tsv+-repo%3Anextstrain%2Faugur&type=code) and strip them to only repo-specific entries.

**Comment by jameshadfield** [comment]
Some recent [related comments on slack](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1730154617851079?thread_ts=1730149944.412499&cid=C01LCTT7JNN):

> colours merging ... is at the category level, i.e. if you set _any_ colour for "region" then no defaults are used (but, e.g. the defaults for "country" would still be used). We don't tend to notice [this behaviour] much because the only defaults we provide are for [region](https://github.com/nextstrain/augur/blob/c0bccdf9a7e8181215af977f79e7bc4ed74e90bd/augur/data/colors.tsv).

> Lat longs are different - we use a key-value store where the key is a tuple of both the category and the deme name (e.g. `("country", "brazil")`). This means overwrites are done in a more fine-grained way - we can overwrite some countries but keep the defaults for others - and we can also add in missing countries whilst keeping the defaults

**Comment by victorlin** [comment]
The change as described in the issue description is not necessary. Shifting conversation about clarification of behavior to a separate issue: #1714
